lib/net/pool.js

```diff
@@ -2012,6 +2145,147 @@ class Pool extends EventEmitter {
+      await indexer.saveFilterHeader(blockHash, filterHeader, filterHash);
+      previousFilterHeader = filterHeader;
+      await this.chain.saveCFHeaderHeight(blockHeight);
+      blockHeight++;
+      const cFHeaderHeight = await this.chain.getCFHeaderHeight();
+      this.logger.info('CFHeaderHeight: %d', cFHeaderHeight);
+    }
+    if (this.headerChain.tail.height <= stopHeight)
+      this.emit('cfheaders');
+    else {
+      const nextStopHeight = stopHeight + 2000 < this.chain.height
+        ? stopHeight + 2000 : this.chain.height;
+      const nextStopHash = await this.chain.getHash(nextStopHeight);
+      this.getcfheadersStopHash = nextStopHash;
+      peer.sendGetCFHeaders(filterType, stopHeight + 1, nextStopHash);
+    }
+  }
+
+  async handleCFilters(peer, packet) {
+    this.logger.info('Received CFilter packet from %s', peer.hostname());
+    if (!this.options.neutrino) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const blockHash = packet.blockHash;
+    const filterType = packet.filterType;
+    const filter = packet.filterBytes;
+
+    if (filterType !== this.getcfheadersFilterType) {
+      peer.ban();
+      peer.destroy();
+      return;
+    }
+
+    const blockHeight = await this.chain.getHeight(blockHash);
+    const stopHeight = await this.chain.getHeight(this.getcfiltersStopHash);
+
+    assert(blockHeight >= this.getcfiltersStartHeight
+      && blockHeight <= stopHeight);
+
+    const basicFilter = new BasicFilter();
+    const gcsFilter = basicFilter.fromNBytes(filter);
+
+    const indexer = this.getFilterIndexer(filtersByVal[filterType]);
+    const filterHeader = await indexer.getFilterHeader(blockHash);
+    await indexer.saveFilter(blockHash, gcsFilter, filterHeader);
+
+    await this.chain.saveCFilterHeight(blockHeight);
+    const cFilterHeight = await this.chain.getCFilterHeight();
+    this.logger.info('CFilter height: %d', cFilterHeight);
+    this.emit('cfilter', blockHash, gcsFilter);
+    const startHeight = stopHeight + 1;
+    let nextStopHeight;
+    if (cFilterHeight === stopHeight
+      && stopHeight < this.chain.height) {
+      if (startHeight + 1000 < this.chain.height) {
+        nextStopHeight = stopHeight + 1000;
+        const stopHash = await this.chain.getHash(nextStopHeight);
+        this.getcfiltersStartHeight = startHeight;
+        this.getcfiltersStopHash = stopHash;
+        this.peers.load.sendGetCFilters(
+          common.FILTERS.BASIC,
+          startHeight,
+          stopHash
+        );
+      } else {
+        nextStopHeight = this.chain.height;
+        const stopHash = await this.chain.getHash(nextStopHeight);
+        this.getcfiltersStartHeight = startHeight;
+        this.getcfiltersStopHash = stopHash;
+        this.peers.load.sendGetCFilters(
+          common.FILTERS.BASIC,
+          startHeight,
+          stopHash
+        );
+        return;
+      }
+    } else if (cFilterHeight === this.chain.height) {
```

Comment:
Maybe also emit a full event here as we are fully synced in neutrino mode

Reply:
Do we really need it? There is no listener bound to pool's 'full' event

Reply:
Will do this.chain.emit('full')